module/zfs/zfs_vnops.c

```diff
@@ -1309,6 +1315,9 @@ zfs_clone_range(znode_t *inzp, uint64_t *inoffp, znode_t *outzp,
 			break;
 		}
 
+		vfs_ratelimit_data_read(inos, inblksz, size);
+		vfs_ratelimit_data_write(outos, inblksz, size);
```

Comment:
Considering deletes are counted as metadata operations, shouldn't cloning also be a pair of metadata read plus metadata write?

Reply:
IMHO block cloning, at least on FreeBSD, can be accessed only via copy_file_range(2) and there is no guarantee data will be cloned and not copied. I think it is more consistent to charge block cloning as regular copy. I think it is similar in this regard to dedup and NOP writes - there might not be the need to write the data to disk, but the write(2) operation  is always charged the same.

Reply:
It should be good to motivate users to be more efficient.  But I accept your point, even though not completely agree.